File: api/src/main/java/cn/hirrao/javaee/utils/ThreadLocalUtil.java
package cn.hirrao.javaee.utils;

import cn.hirrao.javaee.entity.User;

public class ThreadLocalUtil {
    private static final ThreadLocal<User> THREAD_LOCAL = new ThreadLocal<>();

    public static void set(User value) {
        THREAD_LOCAL.set(value);
    }

    public static User get() {
        return THREAD_LOCAL.get();
    }

    public static void remove() {
        THREAD_LOCAL.remove();
    }
}

File: api/src/main/kotlin/com/hirrao/javaee/utils/ThreadLocalUtil.kt
package com.hirrao.javaee.utils

import com.hirrao.javaee.entity.User

object ThreadLocalUtil {
    private val THREAD_LOCAL = ThreadLocal<User>()


    fun set(value: User) {
        THREAD_LOCAL.set(value)
    }


    fun get(): User {
        return THREAD_LOCAL.get()
    }


    fun remove() {
        THREAD_LOCAL.remove()
    }
}
